libs/entitlement/qa/src/javaRestTest/java/org/elasticsearch/entitlement/qa/EntitlementsTestRule.java

```diff
@@ -40,6 +40,7 @@ class EntitlementsTestRule implements TestRule {
                 "files",
                 List.of(
                     Map.of("path", tempDir.resolve("read_dir"), "mode", "read_write"),
+                    Map.of("path", tempDir.resolve("read_dir").resolve("k8s").resolve("..data"), "mode", "read", "exclusive", true),
```

Comment:
`..data` is marked as exclusive here, it shouldn't ever be visible when testing K8s like file mounts (`mount-0.tmp`-> `..data/mount-0.tmp`-> `..version/mount-0.tmp`)